OpenRA.Game/Network/Connection.cs

```diff
@@ -131,57 +136,65 @@ public void Dispose()
 
 	sealed class NetworkConnection : EchoConnection
 	{
-		TcpClient socket;
-		int clientId;
-		ConnectionState connectionState = ConnectionState.Connecting;
-		Thread t;
+		readonly TcpClient tcp;
+		readonly List<byte[]> queuedSyncPackets = new List<byte[]>();
+		volatile ConnectionState connectionState = ConnectionState.Connecting;
+		volatile int clientId = -1;
```

Comment:
Was working on the theory that anybody attempting to access the client ID before a connection is made and then using the incorrect but otherwise valid ID of 0 would be better off getting an invalid ID of -1 which will hopefully cause them to explode somewhere.

I'll check if my thinking is correct there, and if so make LocalClientId throw instead to be obvious.